src/entities/print-job.entity.ts

```diff
@@ -0,0 +1,68 @@
+import { Column, CreateDateColumn, Entity, ManyToOne, PrimaryGeneratedColumn } from "typeorm";
+import { Printer } from "@/entities/printer.entity";
+
+export type PrintJobStatus = "STARTED" | "FINISHED" | "FAILED";
+
+@Entity()
+export class PrintJob {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @ManyToOne(() => Printer, { onDelete: "SET NULL" })
+  printer?: Printer;
+
+  @Column({ nullable: true })
+  printerId: number | null;
+
+  @Column({ type: "varchar", nullable: true })
+  printerName: string | null;
+
+  @Column({ type: "varchar", nullable: false })
+  fileName: string;
+
+  @CreateDateColumn()
+  createdAt: Date;
+
+  @Column({ type: Date, nullable: true })
+  endedAt: Date | null;
+
+  @Column({ type: "varchar", nullable: true })
+  status: PrintJobStatus | null;
+
+  @Column({ type: "float", nullable: true })
+  progress: number | null;
+
+  @Column({ type: "boolean", default: false })
+  completed: boolean;
+
+  @Column({ type: "varchar", nullable: true })
+  reason: string | null;
+
+  @Column({ type: "int", nullable: true })
+  gcodePrintTimeSeconds: number | null;
+
+  // The actual print time shows whether the printer failed
+  @Column({ type: "float", nullable: true })
+  actualPrintTimeSeconds: number | null;
+
+  @Column({ type: "float", nullable: true })
+  nozzleDiameterMm: number | null;
```

Comment:
Things below here seem brittle: any unit change in a slicer and the properties stop working